app/assets/javascripts/discourse/app/lib/topic-list-tracked-filter.js

```diff
@@ -0,0 +1,45 @@
+import Site from "discourse/models/site";
+import User from "discourse/models/user";
+
+export function hasTrackedFilter(queryParams) {
+  if (!queryParams) {
+    return false;
+  }
+
+  return queryParams.f === "tracked" || queryParams.filter === "tracked";
+}
+
+/**
+ * Logic here needs to be in sync with `TopicQuery#tracked_filter` on the server side. See `TopicQuery#tracked_filter`
+ * or the rational behind this decision.
+ */
+export function isTrackedTopic(topic) {
+  if (topic.category_id) {
+    const categories = Site.current().trackedCategoriesList;
+
+    for (const category of categories) {
+      if (category.id === topic.category_id) {
+        return true;
+      }
+
+      if (
+        category.subcategories &&
+        category.subcategories.some((c) => c.id === topic.category_id)
```

Comment:
```suggestion
        category.subcategories?.some((c) => c.id === topic.category_id)
```

Also does this function work as expected if the topic is in a sub-subcategory that the user tracks? (I'm not sure if the `subcategories` property includes sub-subcategories)

Reply:
This is a very good question actually and it actually revealed a bug: https://github.com/discourse/discourse/pull/16963